Improve documentation for building, flashing, troubleshooting

Seneral opened the pull request:
Added alternative methods for building the firmware (linux only),  flashing the firmware (cross-platform), and interacting with the serial port of the firmware (linux).
None of the changes remove or even edit existing documentation.
There's also some generally applicable notes and improvements to troubleshooting problems.

Seneral committed:
Added note that installing Segger J-Link is not required

Seneral committed:
Added manual firmware build instructions for linux

Seneral committed:
Added manual firmware flashing instructions
Also made a clear delineation between boards using adafruit bootloader and dongles using nordic bootloader.

Seneral committed:
Added note for alternatives to "Serial Terminal" of nRF Connect for Desktop
This is relevant because that tool may not work on some linux distros like arch

Seneral committed:
Extended detail on error blink patterns

Seneral committed:
Notes about improving logging as a troubleshooting step

Depact commented on src/diy/smol-slime.md:
```diff
@@ -129,7 +130,7 @@ git clone --single-branch --recurse-submodules -b master https://github.com/Slim
 **Note:** It is recommended to clone to a filepath without whitespaces and/or unicode characters. You may encounters errors when building the firmware.
 
-### Building firmware
+### Building firmware using nRF Connect for VS
```
```suggestion
### Building firmware using nRF Connect for VS Code
```

Depact commented on src/diy/smol-slime.md:
```diff
@@ -104,10 +104,11 @@ Battery must be 50ma or larger with XIAO nRF52840 and 100ma or larger with Super
+    * NOTE: Installing Segger J-Link is not required for known boards.
 * <a href="https://code.visualstudio.com/download" target="_blank">Visual Studio Code</a>
-    * nRF Connect for VS (Install within VS Code Extension tab)
+    * nRF Connect for VS (Recommended for building, Install within VS Code Extension tab)
```
Does it make sense to link official install gude? https://youtu.be/EAJdOqsL9m8
This guide recommend to use `nordic-semiconductor.nrf-connect-extension-pack`, but you only say to install `nordic-semiconductor.nrf-connect`. Is it intentional?

Seneral replied:
I honestly don't know because I haven't been able to get them to work at all (and since I use Code OSS, I had to go through quite a few hoops to even download these extensions)

Depact replied:
Hmm, sounds worth discussing in discord.

Depact commented on src/diy/smol-slime.md:
```diff
@@ -104,10 +104,11 @@ Battery must be 50ma or larger with XIAO nRF52840 and 100ma or larger with Super
+    * Toolchain Manager (Inside nRF Connect; convenient way to build firmware for receiver and tracker)
         * 2.9.0 (Inside Toolchain Manager) Don't use a newer version!
+    * NOTE: Installing Segger J-Link is not required for known boards.
 * <a href="https://code.visualstudio.com/download" target="_blank">Visual Studio Code</a>
```
Later down the doc you refer to `Visual Studio Code` as `VS Code`. I understand it. That's common way to refer to it. But can I suggest note that name irreguality?

Depact commented on src/diy/smol-slime.md:
```diff
@@ -104,10 +104,11 @@ Battery must be 50ma or larger with XIAO nRF52840 and 100ma or larger with Super
+    * Toolchain Manager (Inside nRF Connect; convenient way to build firmware for receiver and tracker)
         * 2.9.0 (Inside Toolchain Manager) Don't use a newer version!
+    * NOTE: Installing Segger J-Link is not required for known boards.
 * <a href="https://code.visualstudio.com/download" target="_blank">Visual Studio Code</a>
```
Can we consider moving docs related to building firmware locally to child page due to huge size. 
Also suggest devidinig needed software on two levels. One sufficient for flashing precompiled firmware, and higher level with additional dependencies to build firmware locally.

Seneral replied:
While I agree, that sounds like a bigger endeavour that's above my "pay grade" so to speak
Maybe a separate PR?
I can ofc just put Smol Slime on the top level after DIY Kit Guide or smth and work with that if you ACK that change

Depact replied:
Fully agree.

Depact commented on src/diy/smol-slime.md:
````diff
@@ -405,6 +496,19 @@ Please open a Github Issue for firmware bugs/issues in the corresponding reposit
+    #include <zephyr/logging/log_ctrl.h>
+    ```
+- If you see the logs are cut off at some point, the buffer size may be too small. This has not been fully resolved yet, merely increasing `CONFIG_LOG_BUFFER_SIZE` in `prj.conf` does not seem to work.
+
````
*Created by Shine Bright ✨, [Depact](https://github.com/Depact), and [Seneral](https://github.com/Seneral).*

Suggestion to add change of last line to this PR like this. Tradition.

Seneral committed:
Fix link convention to open a new tab

Seneral committed:
Clearly separating software for building and flashing
Expanded explanation of use
Expanded setup for VS Code extension
Fix missing assumption in compile command block

Seneral committed:
Improved firmware setup instructions and structure
Separated Firmware Setup into it's own section
Expanded explanation of accessing the firmwares serial console
- Added PuTTY as an alternative to use on windows

Seneral commented:
With this, apart from the restructuring of the page into subpages, I've applied all suggestions
I'd prefer the restructuring to be a separate PR but I'm willing to do it after some further discussion around that

sctanf committed:
Update smol-slime.md

general changes to some wording and layout, rename and add some sections for clarity; additional changes listed

mention nrf54L soc

add descriptions for receiver selection

rewrite dongle description, remove mentions of modifying receiver antenna

make it clear that ebyte dongle is available at aliexpress or alibaba

remove unnecessary detail in BT5040U vs BT5040UA

note that nordic dongle is official dev hardware

explain microcontroller antennas are not optimized and consolidate using same boards as receivers

move antenna modification to supermini only, and explain what it is

add possible search descriptions for supermini

add where to get XIAO board

add basic parts description to tracker

remove bmi warning

consolidate Not Tested notices

rewrite IMU Modules sections

do not describe the imus and drift for imu modules, this should be done from IMU Comparison

describe actually what happens with button configurations

describe when a switch should be used

add a table for battery selection

specify SuperMini for schematics

remove button methods on receiver

remove unnecessary receiver Pair step on firmware update

clarify led blinks

put HID protocol under details

remove slightly condescending wording about making issues

use correct names from firmware

use correct name and author for hardware

remove redundant community labels

ShineBrightMeow committed:
Merge branch 'Test2' into nrf-doc

ShineBrightMeow committed:
Merge branch 'nrf-doc' into nrf-doc
